### PR TITLE
Extend off-chain metadata endpoint

### LINKED ISSUE #1325: Extend off-chain metadata endpoint
### Description

Extend the `/api/v3/blockchain/apps/meta` endpoint.

### Acceptance Criteria

- `isDefault: boolean` based request query params is supported
  - The conditions specified [here](https://docs.google.com/document/d/1fId3uEv1xLfXxBI3sipIGPWl4YiN9caheuQIRZ68VVk/edit#heading=h.vjgtbmdzup88) are respected
- integration tests are up-to-date
- Swagger is up-to-date

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1325 

### How was it solved?

- [x] isDefault: boolean based request query params is supported
- [x] Get application metadata implementation is updated accordingly
- [x] integration tests are up-to-date
- [x] Swagger is up-to-date

### How was it tested?
Local

### CODE REVIEW

File: services/gateway/apis/http-version3/methods/blockchainAppsMetadata.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ module.exports = {
 	params: {
 		chainName: { optional: true, type: 'string', min: 1, max: 20, pattern: regex.NAME },
 		chainID: { optional: true, type: 'string', pattern: regex.CHAIN_ID_CSV },
+		isDefault: { optional: true, type: 'boolean', min: 1, pattern: regex.BOOLEAN },
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/methods/blockchainAppsMetadata.js:29]: We do not really need a patter check for the `boolean` type. Instead, optionally, enforce the `convert` property to `false`, which also happens to be the default value.

Ref: https://github.com/icebob/fastest-validator#boolean

```suggestion
		isDefault: { optional: true, type: 'boolean', convert: false },
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-app-registry/shared/metadata.js [hunk 2]
```diff
@@ -143,13 +143,15 @@ const getBlockchainAppsMetadata = async (params) => {
 	}
 
 	const limit = params.limit * config.supportedNetworks.length;
-	const defaultApps = await applicationMetadataTable.find(
-		{ ...params, limit, isDefault: true },
-		['network', 'appDirName'],
-	);
-	blockchainAppsMetadata.data = defaultApps;
+	if (!('isDefault' in params) || params.isDefault) {
```
Comment (priojeetpriyom) [inline services/blockchain-app-registry/shared/metadata.js:146]: Alternatively, could we write like below?

```suggestion
	if (params.isDefault !== false) {
```

File: services/blockchain-app-registry/shared/metadata.js [hunk 3]
```diff
@@ -143,13 +143,15 @@ const getBlockchainAppsMetadata = async (params) => {
...
+		const defaultApps = await applicationMetadataTable.find(
+			{ ...params, limit, isDefault: true },
+			['network', 'appDirName'],
+		);
+		blockchainAppsMetadata.data = defaultApps;
+	}
 
-	if (defaultApps.length < params.limit) {
+	if (!params.isDefault && blockchainAppsMetadata.data.length < params.limit) {
```
Comment (priojeetpriyom) [inline services/blockchain-app-registry/shared/metadata.js:154]: Alternatively, could we write like below?

```suggestion
	if (params.isDefault !== true && blockchainAppsMetadata.data.length < params.limit) {
```

**Review by priojeetpriyom — APPROVED** [review]

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**